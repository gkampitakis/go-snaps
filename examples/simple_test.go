package examples

import (
	"bytes"
	"testing"

	"github.com/gkampitakis/go_snaps/snaps"
)

func TestSimple(t *testing.T) {
	t.Run("should make an int snapshot", func(t *testing.T) {
		snaps.MatchSnapshot(t, 5)
	})

	t.Run("should make a string snapshot", func(t *testing.T) {
		snaps.MatchSnapshot(t, "string snapshot")
	})

	t.Run("should make a map snapshot", func(t *testing.T) {
		m := map[string]interface{}{
			"mock-0": "value",
			"mock-1": 2,
			"mock-2": func() {},
			"mock-3": float32(10.4),
		}

		snaps.MatchSnapshot(t, m)
	})

	t.Run("should make multiple entries in snapshot", func(t *testing.T) {
		snaps.MatchSnapshot(t, 5, 10, 20, 25)
	})

	t.Run("should make create multiple snapshot", func(t *testing.T) {
		snaps.MatchSnapshot(t, 1000)
		snaps.MatchSnapshot(t, "another snapshot")
		snaps.MatchSnapshot(t, `{
			"user": "gkampitakis",
			"id": 1234567,
			"data": [ ]
		}`)
	})

	t.Run("nest", func(t *testing.T) {
		t.Run("more", func(t *testing.T) {
			t.Run("one more nested test", func(t *testing.T) {
				snaps.MatchSnapshot(t, "it's okay")
			})
		})
	})
}

func TestSimpleTable(t *testing.T) {
	type testCases struct {
		description string
		input       interface{}
	}

	for _, scenario := range []testCases{
		{
			description: "string",
			input:       "input",
		},
		{
			description: "integer",
			input:       10,
		},
		{
			description: "map",
			input: map[string]interface{}{
				"test": func() {},
			},
		},
		{
			description: "buffer",
			input:       bytes.NewBufferString("Buffer string"),
		},
	} {
		t.Run(scenario.description, func(t *testing.T) {
			snaps.MatchSnapshot(t, scenario.input)
		})
	}
}

func TestSimpleConfig(t *testing.T) {
	s := snaps.New(snaps.SnapsDirectory("mySnaps"), snaps.SnapsExtension("txt"))

	s.MatchSnapshot(t, 10)
}
