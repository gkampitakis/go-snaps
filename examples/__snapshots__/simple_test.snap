
[TestSimple/should_make_an_int_snapshot - 1]
int(5)
---

[TestSimple/should_make_a_string_snapshot - 1]
string snapshot
---

[TestSimple/should_make_a_map_snapshot - 1]
map[string]interface {}{
    "mock-0": "value",
    "mock-1": int(2),
    "mock-2": func() {...},
    "mock-3": float32(10.399999618530273),
}
---

[TestSimple/should_make_multiple_entries_in_snapshot - 1]
int(5)
int(10)
int(20)
int(25)
---

[TestSimple/should_make_create_multiple_snapshot - 1]
int(1000)
---

[TestSimple/should_make_create_multiple_snapshot - 2]
another snapshot
---

[TestSimple/should_make_create_multiple_snapshot - 3]
{
            "user": "gkampitakis",
            "id": 1234567,
            "data": [ ]
        }
---

[TestSimpleTable/string - 1]
input
---

[TestSimpleTable/integer - 1]
int(10)
---

[TestSimpleTable/map - 1]
map[string]interface {}{
    "test": func() {...},
}
---

[TestSimple/nest/more/one_more_nested_test - 1]
it's okay
---

[TestSimpleTable/buffer - 1]
&bytes.Buffer{
    buf:      {0x42, 0x75, 0x66, 0x66, 0x65, 0x72, 0x20, 0x73, 0x74, 0x72, 0x69, 0x6e, 0x67},
    off:      0,
    lastRead: 0,
}
---
